contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol

```diff
@@ -374,13 +373,18 @@ abstract contract FunctionsBilling is HasRouter, IFunctionsBilling {
       commitment
     );
 
+    FulfillResult fulfillResult = FulfillResult(result);
+    if (fulfillResult == FulfillResult.USER_SUCCESS || fulfillResult == FulfillResult.USER_ERROR || fulfillResult == FulfillResult.SUBSCRIPTION_BALANCE_INVARIANT_VIOLATION) {
+      delete s_requestCommitments[requestId];
+    }
+
     // Reimburse the transmitter for the fulfillment gas cost
     s_withdrawableTokens[msg.sender] = gasOverheadJuels + callbackCostJuels;
```

Comment:
Lines 381-385 should only happen on FulfillResult.USER_SUCCESS  or FulfillResult.USER_ERROR
Could you add branching logic for that?